File: CoderCave/Models/Inquire.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace CoderCave.Models
{
    public class Inquire
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public string Title { get; set; }
        public string Content { get; set; }
        public string ContentSummary { get; set; }
        public DateTime CreatedAt { get; set; }
        public User User { get; set; }
        public string AuthorName { get; set; }
        public string AuthorImageURL { get; set; }
        public List<Answer> Answers { get; set; } = new List<Answer>();
        public int AnswersCount { get; set; }
        public List<InquireComment> Comments { get; set; } = new List<InquireComment>();
        public int CommentsCount { get; set; }
        public List<Tag> Tags { get; set; } = new List<Tag>();
        public int VotesCount { get; set; }
        public int Score { get; set; }
        public bool IsArchived { get; set; }
        public bool AnswerSelected { get; set; }
    }
}

File: CoderCave/client/src/components/inquire/Inquire.js
import { useContext, useEffect } from "react";
import { Link, useLocation } from "react-router-dom";
import { Card, CardBody, CardText, CardTitle } from "reactstrap";
import { SearchContext } from "../../providers/SearchProvider";


const Inquire = ({ inquire }) => {

  // const location = useLocation();

  // const { searchTerm } = useContext(SearchContext);

  // useEffect(() => {

  //   const onSearchResults = location.pathname.includes('search');

  //   if (onSearchResults) {
  //     console.log(searchTerm)
  //   }
  // }, []);

  return (
    <Card className="mt-2">
      <CardBody>
        <CardTitle tag="h5"><Link to={`/inquire/details/${inquire.id}`}>{inquire.title}</Link> { inquire.answerSelected && <i className="selected-answer bi bi-check-lg"></i> }</CardTitle>
        <CardText>{inquire.contentSummary}</CardText>
      </CardBody>
      <CardBody>
        <CardText>
          Posted By: <Link to={`/user/details/${inquire.userId}`}>{inquire.authorName}</Link>&nbsp;on {new Date(inquire.createdAt).toLocaleString()}
        </CardText>
        <CardText>
          Answers: {inquire.answersCount} &nbsp;
          Comments: {inquire.commentsCount} &nbsp;
          Votes: {inquire.votesCount}
        </CardText>
      </CardBody>
    </Card>
  );
};

export default Inquire;